### PR TITLE
Add helper methods to StackFrame

### PR DESCRIPTION
**Pull request by kashike**
[SpongeAPI](https://github.com/SpongePowered/SpongeAPI/pull/1649) | [**SpongeCommon**](https://github.com/SpongePowered/SpongeCommon/pull/1531) | [SpongeForge](https://github.com/SpongePowered/SpongeForge/pull/1772)

### CODE REVIEW

File: src/main/java/org/spongepowered/common/event/SpongeCauseStackManager.java [hunk 1]
```diff
@@ -275,8 +277,8 @@ public void popCauseFrame(StackFrame oldFrame) {
     public static class CauseStackFrameImpl implements StackFrame {
 
         // lazy loaded
-        private Map<EventContextKey<?>, Object> stored_ctx_values;
-        private Set<EventContextKey<?>> new_ctx_values;
+        @Nullable private Map<EventContextKey<?>, Object> stored_ctx_values;
+        @Nullable private Set<EventContextKey<?>> new_ctx_values;
```
Comment (liach) [inline src/main/java/org/spongepowered/common/event/SpongeCauseStackManager.java:281]: Just curious, why are variables in snake case here?

Comment (kashike) [reply src/main/java/org/spongepowered/common/event/SpongeCauseStackManager.java:281]: @Deamon5550 can tell you why

Comment (Deamon5550) [reply src/main/java/org/spongepowered/common/event/SpongeCauseStackManager.java:281]: Habit

Comment (liach) [reply src/main/java/org/spongepowered/common/event/SpongeCauseStackManager.java:281]: Well, this looks way to c++, which is in violation of Sponge code style. I think you can safely change them as they are not exposed in the API.

Comment (kashike) [reply src/main/java/org/spongepowered/common/event/SpongeCauseStackManager.java:281]: I'm going to leave them as-is for now.

**Review by stephan-gh — APPROVED** [review]

**Comment by JBYoshi** [comment]
@kashike The API PR was merged, but this and the SpongeForge PR are still open. Could you merge these?

**Comment by JBYoshi** [comment]
Also the SpongeVanilla PR. https://github.com/SpongePowered/SpongeVanilla/pull/350

**Comment by kashike** [comment]
These aren't going to be merged.

**Comment by JBYoshi** [comment]
The API one was, though. It's giving me some compilation errors for unimplemented methods.

**Comment by liach** [comment]
Are you going to commit directly instead?

**Pull request closed without merging**